### PR TITLE
macOS titlebar style settings

### PR DESCRIPTION
**Pull request by EnricoMonese**
schermata 2017-03-26 alle 16 24 51

Settings to select titlebar style on macOS
a
- First one is default macOS
- Custom titlebar, top 30px act as draggable region
- Custom titlebar with inset _traffic light_ buttons, window is draggable from everywhere

When entering fullscreen the layout goes back to normal and fills the space of the window buttons.

### CODE REVIEW

**Comment by EnricoMonese** [comment]
What would be the best way to make this setting only appear on macOS?

**Comment by antelle** [comment]
Add an option to FeatureDetector: supporsTitleBarStyles, check for macos there.
Probably we could implement this for Windows later as well.

**Comment by antelle** [comment]
Apart from this, looks cool, I'll prettify settings in electron app reading and merge it.
Thank you! 🎁

File: app/scripts/views/settings/settings-general-view.js [hunk 1]
```diff
@@ -82,7 +82,10 @@ const SettingsGeneralView = Backbone.View.extend({
         const updateFound = UpdateModel.instance.get('updateStatus') === 'found';
         const updateManual = UpdateModel.instance.get('updateManual');
         const storageProviders = this.getStorageProviders();
+        const platform = (typeof process !== 'undefined') ? process.platform : 'web';
+
```
Comment (antelle) [inline app/scripts/views/settings/settings-general-view.js]: please don't do this

File: app/templates/settings/settings-general.hbs [hunk 2]
```diff
@@ -64,6 +64,7 @@
             <option value="2" {{#ifeq fontSize 2}}selected{{/ifeq}}>{{res 'setGenFontSizeLargest'}}</option>
         </select>
     </div>
+    {{#ifeq platform 'darwin'}}
```
Comment (antelle) [inline app/templates/settings/settings-general.hbs]: should be in feature-detector.js

File: app/scripts/views/settings/settings-general-view.js (hunk 1 above)
Comment (EnricoMonese) [inline app/scripts/views/settings/settings-general-view.js]: Just saw your suggestion about the FeatureDetector 👍

Comment (antelle) [inline app/scripts/views/settings/settings-general-view.js]: :+1: cool

**Comment by EnricoMonese** [comment]
Added it to `FeatureDetector: supporsTitleBarStyles` but it doesn't seem to work, what did I miss?

**Comment by antelle** [comment]
You added it as a function, to it should be: supportsTitleBarStyles()

**Comment by EnricoMonese** [comment]
Ah silly me, fixed it!

**Comment by EnricoMonese** [comment]
Just realized that I made the pr for the master branch, should it be changed to the dev branch?

**Comment by antelle** [comment]
yes, it's better to check on develop: some things have been changed

**Pull request merged**